frodo.js

```diff
@@ -1,45 +1,49 @@
 /**
```

Comment:
SCRIPT-Well done with the efficient error handling, it covers all the possible scenarios.